ui/StatusQ/include/StatusQ/objectproxymodel.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <QIdentityProxyModel>
+#include <QPointer>
+#include <QQmlContext>
+#include <QQmlPropertyMap>
+#include <QSet>
+
+#include "modelsyncedcontainer.h"
+
+class QQmlComponent;
+class QQmlEngine;
+
+class ObjectProxyModel : public QIdentityProxyModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QQmlComponent* delegate READ delegate
+               WRITE setDelegate NOTIFY delegateChanged)
+
+    Q_PROPERTY(QStringList expectedRoles READ expectedRoles
+               WRITE setExpectedRoles NOTIFY expectedRolesChanged)
+
+    Q_PROPERTY(QStringList exposedRoles READ exposedRoles
+               WRITE setExposedRoles NOTIFY exposedRolesChanged)
+
+public:
+    explicit ObjectProxyModel(QObject* parent = nullptr);
+
+    QVariant data(const QModelIndex& index, int role) const override;
+    void setSourceModel(QAbstractItemModel* sourceModel) override;
+    QHash<int, QByteArray> roleNames() const override;
+
+    QQmlComponent* delegate() const;
+    void setDelegate(QQmlComponent* delegate);
+
+    void setExpectedRoles(const QStringList& expectedRoles);
+    const QStringList& expectedRoles() const;
+
+    void setExposedRoles(const QStringList& exposedRoles);
+    const QStringList& exposedRoles() const;
+
+    Q_INVOKABLE QObject* proxyObject(int index);
+    const QObject* proxyObject(int index) const;
+
+signals:
+    void delegateChanged();
+    void expectedRolesChanged();
+    void exposedRolesChanged();
+
+protected slots:
+    void resetInternalData();
```

Comment:
Good question. I'm doing cleanup here according to https://doc.qt.io/qt-5/qabstractitemmodel.html#resetInternalData Starting from Qt 6 it's virtual, but not in Qt 5 (https://codebrowser.dev/qt5/qtbase/src/corelib/itemmodels/qabstractitemmodel.h.html#QAbstractItemModel). I presume thanks to qmoc  it works as virtual anyways because it's called on model reset, even thought the signal is emitted from base class. At least tests confirm that.

Reply:
Yeah, I thought they forgot to add the `virtual` here! Thanks for confirming that :)